tests/integration-tests/tests/networking/test_security_groups.py

```diff
@@ -72,9 +72,25 @@ def test_overwrite_sg(region, scheduler, custom_security_group, pcluster_config_
         fsx_security_groups = ec2_client.describe_network_interfaces(NetworkInterfaceIds=[network_interface_id])[
             "NetworkInterfaces"
         ][0]["Groups"]
-        logging.info("Asserting the network interface of FSx has and only has the custom security group")
-        assert_that(fsx_security_groups[0]["GroupId"]).is_equal_to(custom_security_group_id)
+        logging.info(
+            "Asserting the network interface of FSx has only the dedicated security group "
+            "allowing all traffic to itself and to the custom security group"
+        )
         assert_that(fsx_security_groups).is_length(1)
+        fsx_security_group = fsx_security_groups[0]
+        fsx_security_group_id = fsx_security_group["GroupId"]
+        # We expect the FSx Security Group to have exactly 4 rules:
+        #  * 2 rules (ingress and egress) allowing traffic to/from the FSx Security Group itself
+        #  * 2 rules (ingress and egress) allowing traffic to/from the custom Security Group
+        _assert_security_group_rules(
+            ec2_client,
+            fsx_security_group_id,
+            [
```

Comment:
This loop might be moved into the `_assert_security_group_rules` function and you can just pass the two security groups to loop for.

The `_assert_security_group_rules` test is very specific to egress so we can move the entire logic there and rename it to `_assert_egress_security_group_rules`.

Reply:
> This loop might be moved into the _assert_security_group_rules function and you can just pass the two security groups to loop for.

I agree, thank you.

> The _assert_security_group_rules test is very specific to egress so we can move the entire logic there and rename it to _assert_egress_security_group_rules.

The assertion covers both egress (IsEgress: True) and ingress (IsEgress: False), so I'd keep the current name.